dvc/scm/git/backend/dulwich.py

```diff
@@ -399,9 +399,11 @@ def push_refspec(
             ) from exc
 
         def update_refs(refs):
+            from dulwich.objects import ZERO_SHA
+
             new_refs = {}
             for ref, value in zip(dest_refs, values):
-                if ref in refs:
+                if ref in refs and value != ZERO_SHA:
```

Comment:
Delete something is also a kind of overwrite. Maybe 

```
if ref in refs:
    local_sha = value
```

Here and add a `--force` flag to it?

is better?

Reply:
If you are explicitly deleting a remote ref, the assumption is that it already exists, and that you want to delete it. I don't see any reason to need a separate `-f/--force` in this case. (This is the same way that `git push -d` works - you don't need to do `git push -f -d`)

Reply:
Also, could you clarify why the `and value != ZERO_SHA` check is needed here?

Reply:
And here what I suggest is that it is better to error out here in `push_refspec` because the `ZERO_SHA` is overwritten to the remote. But we can add the `force=True` flag in the call 

```
repo.scm.push_refspec(remote, None, ref_name)
```
I agreed that we don't need to use `--force` flag here. all of the operations in `dvc exp remove` are dangerous, do not need to remind users.

Reply:
Pushing the `None` source refspec is an explicit remove operation, it shouldn't require an additional `force=True`.

Reply:
> I agreed that we don't need to use `--force` flag here. all of the operations in `dvc exp remove` are dangerous, do not need to remind users.

In general, it might be helpful to list the refs found for removal and ask for confirmation by default. That's true for all of `dvc exp remove`, not just with `--git-remote`. I'll open a separate issue for that.